torch/nn/_functions/thnn/upsampling.py

```diff
@@ -62,15 +59,39 @@ def backward(self, grad_output):
         return grad_input
 
 
-class UpsamplingBilinear2d(_UpsamplingBase):
+class UpsamplingBilinear2d(Function):
+
+    def __init__(self, size=None, scale_factor=None):
+        super(UpsamplingBilinear2d, self).__init__()
+        if size is None and scale_factor is None:
+            raise ValueError('either size or scale_factor should be defined')
+        if scale_factor is not None:
+            if not isinstance(scale_factor, (Integral, tuple)):
+                raise ValueError('scale_factor must be a non-negative integer, or a tuple of non-negative integers')
+            if isinstance(scale_factor, tuple):
+                try:
+                    assert len(scale_factor) == 2
+                    for i in scale_factor:
+                        assert isinstance(i, Integral)
+                        assert i >= 1
+                except AssertionError as e:
+                    raise ValueError('scale_factor must be a non-negative integer, or a tuple of non-negative integers')
+                
+        if size is not None and not isinstance(size, tuple):
+            size = (size, size)
+        if scale_factor is not None and not isinstance(scale_factor, tuple):
+            scale_factor = (scale_factor, scale_factor)
```

Comment:
That's what a `_pair` helper from `_utils` is for.

Also, if you do that before the type checks, you can get rid of all `isinstance(scale_factor, tuple)` checks, because you can be sure that it is a tuple now.